experimental/dds/tree2/src/test/shared-tree/fuzz/targetedFuzz.spec.ts

```diff
@@ -120,35 +37,100 @@ export async function performFuzzActionsComposeVsIndividual(
 		setPayload: 1,
 	};
 	describe("Anchors are unaffected by aborted transaction", () => {
-		runFuzzBatch(
-			makeOpGenerator,
-			performFuzzActionsAbort,
-			opsPerRun,
-			runsPerBatch,
-			random,
-			editGeneratorOpWeights,
-		);
+		const generatorFactory = () =>
+			takeAsync(opsPerRun, makeOpGenerator(editGeneratorOpWeights));
+		const generator = generatorFactory() as AsyncGenerator<TreeOperation, AbortFuzzTestState>;
+		const model: DDSFuzzModel<
+			SharedTreeTestFactory,
+			Operation,
+			DDSFuzzTestState<SharedTreeTestFactory>
+		> = {
+			workloadName: "SharedTree",
+			factory: new SharedTreeTestFactory(onCreate),
+			generatorFactory: () => generator,
+			reducer: fuzzReducer,
+			validateConsistency: () => {},
+		};
+
+		const emitter = new TypedEventEmitter<DDSFuzzHarnessEvents>();
+		emitter.on("testStart", (initialState: AbortFuzzTestState) => {
+			// building the anchor for anchor stability test
+			const cursor = initialState.clients[0].channel.forest.allocateCursor();
+			moveToDetachedField(initialState.clients[0].channel.forest, cursor);
+			cursor.enterNode(0);
+			cursor.getPath();
+			cursor.firstField();
+			cursor.getFieldKey();
+			cursor.enterNode(1);
+			initialState.firstAnchor = cursor.buildAnchor();
+			cursor.free();
+			initialState.clients[0].channel.transaction.start();
+		});
+
+		emitter.on("testEnd", (finalState: AbortFuzzTestState) => {
+			// aborts any transactions that may still be in progress
+			finalState.clients[0].channel.transaction.abort();
+			validateTree(finalState.clients[0].channel, [initialTreeState]);
+			// validate anchor
+			const expectedPath: UpPath = {
+				parent: {
+					parent: undefined,
+					parentIndex: 0,
+					parentField: rootFieldKeySymbol,
+				},
+				parentField: brand("foo"),
+				parentIndex: 1,
+			};
+			assert(finalState.firstAnchor !== undefined);
+			const anchorPath = finalState.clients[0].channel.locate(finalState.firstAnchor);
+			assert(compareUpPaths(expectedPath, anchorPath));
+		});
+
+		createDDSFuzzSuite(model, {
+			defaultTestCount: runsPerBatch,
+			numberOfClients: 1,
+			emitter,
+		});
 	});
 	const composeVsIndividualWeights: Partial<EditGeneratorOpWeights> = {
 		setPayload: 1,
 		insert: 1,
 		delete: 1,
 	};
 	describe("Composed vs individual changes converge to the same tree", () => {
-		runFuzzBatch(
-			makeOpGenerator,
-			performFuzzActionsComposeVsIndividual,
-			opsPerRun,
-			runsPerBatch,
-			random,
-			composeVsIndividualWeights,
-		);
+		const generatorFactory = () =>
+			takeAsync(opsPerRun, makeOpGenerator(composeVsIndividualWeights));
+		const model: DDSFuzzModel<
+			SharedTreeTestFactory,
+			Operation,
+			DDSFuzzTestState<SharedTreeTestFactory>
+		> = {
+			workloadName: "SharedTree",
+			factory: new SharedTreeTestFactory(onCreate),
+			generatorFactory,
+			reducer: fuzzReducer,
+			validateConsistency: () => {},
+		};
+		const emitter = new TypedEventEmitter<DDSFuzzHarnessEvents>();
```

Comment:
this test is equivalent to the previous version of the code, so no need to fix anything as part of this PR, but I'm wondering--what's the intent of this targeted test? Is the entire set of fuzz actions run within a single transaction? if so, how's that enforced?

Reply:
I think this test is for checking that a set of changes applied to it individually vs applying them as a composed change. I believe that the way our transaction works is that when we call `transaction.start()` all of the changes that are applied to it 1 by 1, and when we call `transaction.commit()`, we roll back those changes, compose them, then apply them again (or the other way around).

Which is why we store the state of the tree in `treeViewBeforeCommit` before calling `transaction.commit` to see if it's equivalent to the tree after calling `transaction.commit`

Reply:
gotcha. and I see now that the default weight for start/end transaction is 0. Might be worth explicitly disabling the transaction-level weights in `composeVsIndividualWeights`, in case that default ever changes. But feel free to do as part of a later PR.

Reply:
Yes, that makes sense! I will create a new PR adjusting the weights :)